### PR TITLE
Add jax.scipy.stats.binom

### LINKED ISSUE #16550: jax.scipy.stats.binom missing
I can't find an implementation of the jax equivalent of the binomial from scipy.stats. Specifically in my case I am looking for a jittable way to calculate the binomial cdf i.e. `jax.scipy.stats.binom.cdf(k, n, p)`

### PR DESCRIPTION
**Pull request by jakevdp**
Adds `pmf` and `logpmf`; Fixes #16550

### CODE REVIEW

**Comment by jakevdp** [comment]
CI failure is unrelated; seems to be due to https://github.com/tensorflow/tensorflow/pull/56908#issuecomment-1608926955

**Review by froystig — APPROVED** [review]

File: jax/_src/scipy/stats/binom.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+@_wraps(osp_stats.nbinom.logpmf, update_doc=False)
+def logpmf(k: ArrayLike, n: ArrayLike, p: ArrayLike, loc: ArrayLike = 0) -> Array:
+    """JAX implementation of scipy.stats.binom.logpmf."""
+    k, n, p, loc = promote_args_inexact("binom.logpmf", k, n, p, loc)
+    y = lax.sub(k, loc)
+    comb_term = lax.sub(
+        gammaln(n + 1),
+        lax.add(gammaln(y + 1), gammaln(n - y + 1))
+    )
```
Comment (axch) [inline jax/_src/scipy/stats/binom.py:33]: This semantically correct, but numerically unfortunate for large-ish `n` and `y` close to either `n` or 0.  To wit, if `y` is close to 0, `gammaln(y + 1)` will be small, and this expression will produce a catastrophic cancellation between `gammaln(n + 1)` and `gammaln(n - y + 1)`.  Unfortunately, the solution is somewhat involved -- TFP had to reimplement the `lbeta` function in order to get this right: see [implementation](https://github.com/tensorflow/probability/blob/main/tensorflow_probability/python/math/special.py#L1854-L2129) and [usage](https://github.com/tensorflow/probability/blob/main/tensorflow_probability/python/distributions/binomial.py#L580-L582).  I also expect that code to be slower to compile and execute than what we have here.

I'm not sure of the right thing to do here.  We could
- Add a warning to the documentation that JAX prioritizes speed over corner-case accuracy, and this code will be somewhat inaccurate for large `n` and `y` close to `n` or 0; we can also direct users to TFP for the more accurate but presumably slower version.
- Fork TFP's implementation of `lbeta` and use it here.
- I assume we don't want to take TFP's `lbeta` as a dependency.

**Comment by carlosgmartin** [comment]
@jakevdp Are `jax.scipy.stats.binom` and `jax.scipy.stats.nbinom` missing from [the documentation](https://jax.readthedocs.io/en/latest/jax.scipy.html)?

**Comment by jakevdp** [comment]
Thanks for the note - Yes, it looks like they're missing. I'll add them!
